### ISSUE TITLE
Angular 2 Module

### ISSUE THREAD

**Issue by sebastianhaas** [opensheetmusicdisplay/opensheetmusicdisplay#106]
Create an Angular 2 module that is

- installable over npm,
- exports opensheetmusicdisplay as an Angular 2 component
- takes xml/mxl data as `@Input`

**Comment by bohoffi** [comment]
IMHO the process to use OSMD right now in an Angular application isn't a very big deal. One could easily create a custom component and include all the stuff required to get OSMD to work. So IMHO I don't see the requirement to make OSMD available as an Angular module.

But 1 scenario could be that the module import (`OSMDModule.forRoot()`) takes some kind of config.

Could you explain your thoughts on the PROs using OSMD as an Angular module?

**Comment by sebastianhaas** [comment]
> OSMD right now in an Angular application isn't a very big deal.

I agree.

> One could easily create a custom component and include all the stuff required to get OSMD to work.

I agree too.

The reason/motivation for having an Angular (2.x-4.x) module is simply to address a greater number of people and to lower entry barriers. I think that, once set up, maintaining such a module does not cause a big workload plus it might be useful for me to discover (and resolve) challenges when using OSMD withing Angular applications before others have to do so :) The same argumentation goes for the wordpress plugin #25.

However, as we advance further with this project, by using ES6 imports Angular might be able to remove not required parts of OSMD during bundling (tree shaking). The same goes for AOT compiled modules. Modules ready for AOT compiling need to be published with some fancy module descriptors, so you need to compile stuff with ngc rather than tsc etc. so Angular might be able to optimize OSMD, but that probably requires a certain setup.
I just thought it wouldn't hurt to have it done once in a centralized manner, rather than lots of people having to do it on their own stumbling over the same problems over and over again.

Obviously non of this would work out-of-the-box right now, so you might be right, apart from saving  Angular users some of their time, there's no real benefit.

**Comment by bohoffi** [comment]
Ok, your argumentation is good.

If I can support you just let me know :)

**Comment by eamador** [comment]
@sebastianhaas what's the state of this? Is there any development in progress where i can contribute?

**Comment by sebastianhaas** [comment]
No, currently there isn't any development in progress regarding this feature. I think it would make proper sense to outsource this to another repository though.

I've created https://github.com/sebastianhaas/opensheetmusicdisplay-angular for this, and I would welcome PRs on the matter. :)

I'd vote for https://github.com/trekhleb/angular-library-seed as a starting point- it provides all the essential dev setup as far as I can tell. It should be easy to create the required module then, just replace the default TickTock from the sample repo with some OSMD boilerplate. Since OSMD is written in Typescript as well, everything should be really easy to do.
